src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/IProvenBlockHeaderRepository.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NBitcoin;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.ProvenBlockHeaders
+{
+    /// <summary>
+    /// Interface for database <see cref="ProvenBlockHeader"></see> repository.
+    /// </summary>
+    public interface IProvenBlockHeaderRepository : IDisposable
+    {
+        /// <summary>
+        /// Initializes <see cref="ProvenBlockHeader"/> items database.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        Task InitializeAsync();
+
+        /// <summary>
+        /// Retrieves <see cref="ProvenBlockHeader"/> items from the database.
+        /// </summary>
+        /// <param name="fromBlockHeight">Block height to start range.</param>
+        /// <param name="toBlockHeight">Block height end range.</param>
+        /// <returns>A dictionary of <see cref="ProvenBlockHeader"/> items by block height.</returns>
+        Task<Dictionary<int, ProvenBlockHeader>> GetAsync(int fromBlockHeight, int toBlockHeight);
```

Comment:
The caller already knows the heights of the blocks that will be returned, no need to return a height:header collection. Just proven headers collection

Reply:
Actually just made it return a List so not to worry.

Reply:
I've changed it to return `List<ProvenBlockHeaders>`.